### ISSUE TITLE
Recursive endpoint to get inscription on sat

### ISSUE THREAD

**Issue by casey** [ordinals/ord#2277]
See discussion in #2200. Something like:

```
/sat/N # first inscription on sat N
/sat/N/X # Xth inscription on sat N, oldest first
/sat/N/-X # Xth inscription on sat N, newest first
```

**Comment by onchainguy-eth** [comment]
Inscriptions on the same sat are prohibited by the current ord codebase aren't they? Was looking at [this line](https://github.com/ordinals/ord/blob/fe82d4cc14e169282c62e7a61bad8b70fec84f62/src/subcommand/wallet/inscribe.rs#L178C5-L189C6). Of course, one can just remove the condition, build and inscribe but was wondering what's the idea here.

**Comment by devords** [comment]
Having these same API endpoints available under inscriptions would be super useful too:

```
/inscription/ID/sat/X
//inscription/ID/sat/-X
```

Also, is it necessary to have `/sat/N` when it's always going to be the same as `/sat/N/0`? Instead, `/sat/N.json` endpoint can be used for returning a JSON of sat specific information such as rarity and blockheight.

**Comment by null-ish** [comment]
Looking forward to this so I can make my own "social media wall" sat which I will post stuff to by re-inscribing the "data" sat it's connected to.

`src="/content/sat/DATA_SAT_NUMBER/-1"`

Plenty other use cases become possible with this. Sky is the limit!

**Comment by raphjaph** [comment]
probably not going to add the `/content/sat/...` endpoint since this makes caching more difficult. 

Recursive inscriptions would need to get the id from `/sat/N` and then get the content with that id like so `/content/ID`.

**Comment by devords** [comment]
@raphjaph Having `/content/sat/-1` would be really convenient, especially with HTML inscriptions.

Consider the following inscription:

```html
<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8">
  <link rel="stylesheet" href="/sat/XXX/-1" />
  <script src="/sat/YYY/-1"></script>
</head>

<body>
  ...
</body>
</html>
```

With sat indexed endpoints for fetching content, I can easily update this inscription. Same is certainly possible with `/sat/N` + `/content/ID`, but it'd require a lot more code gymanstics.

**Comment by raphjaph** [comment]
Closed by #2452
